src/App.tsx

```diff
@@ -33,25 +40,62 @@ export const PATHS = {
   WELCOME_PAGE: "/welcome",
 };
 
+const queryClient = new QueryClient({
+  defaultOptions: {
+    mutations: {
+      networkMode: 'online',
+    },
+    queries: {
+      networkMode: 'online',
+      staleTime: 1000 * 20, // 20 seconds
+      gcTime: 1000 * 60 * 60 * 24 * 7, // 1 week
+      retry: 0,
+    }
+  }
+});
+addDefaultMutationFns(queryClient);
+
+const store = new Storage();
+await store.create();
+
+const persister = createAsyncStoragePersister({
+  storage: {
+    getItem: async (key) => store.get(key),
+    setItem: async (key, value) => store.set(key, value),
+    removeItem: async (key) => store.remove(key),
+  },
+});
+
 const App: React.FC = () => (
-  <IonApp>
-    <IonReactRouter>
-      <IonRouterOutlet>
-        <Route exact path={PATHS.WELCOME_PAGE}>
-          <WelcomePage />
-        </Route>
-        <Route exact path={PATHS.TUTORIAL_PAGE}>
-          <TutorialPage />
-        </Route>
-        <Route exact path={PATHS.HOME_PAGE}>
-          <Home />
-        </Route>
-        <Route exact path="/">
-          <Redirect to={PATHS.WELCOME_PAGE} />
-        </Route>
-      </IonRouterOutlet>
-    </IonReactRouter>
-  </IonApp>
+  <PersistQueryClientProvider
+    client={queryClient}
+    persistOptions={{ persister: persister, maxAge: Infinity }}
+    onSuccess={() => {
+      queryClient.resumePausedMutations().then(() => {
+        queryClient.invalidateQueries()
+      })
+    }}
+  >
+    <IonApp>
+      <IonReactRouter>
+        <IonRouterOutlet>
+          <Route exact path={PATHS.WELCOME_PAGE}>
+            <WelcomePage />
+          </Route>
+          <Route exact path={PATHS.TUTORIAL_PAGE}>
+            <TutorialPage />
+          </Route>
+          <Route exact path={PATHS.HOME_PAGE}>
+            <Home />
+          </Route>
+          <Route exact path="/">
+            <Redirect to={PATHS.WELCOME_PAGE} />
+          </Route>
+        </IonRouterOutlet>
+      </IonReactRouter>
+    </IonApp>
+    <ReactQueryDevtools initialIsOpen={false} />
```

Comment:
The docs for the dev tools are here: https://tanstack.com/query/latest/docs/framework/react/devtools

Specifically, the docs say:

> Devtools are excluded in production builds.